### ISSUE TITLE
DefaultImageHeaderParser fails to determine correct orientation

### ISSUE THREAD

**Issue by grote** [bumptech/glide#3944]
**Glide Version**: 4.10.0

**Integration libraries**: none

**Device/Android Version**: Pixel 2 XL Android 10

**Issue details / Repro steps / Use case background**: Glide (but in its defense also my browser) fails to determine the orientation of this image:

![rotate-test](https://user-images.githubusercontent.com/244947/67391990-16dc6780-f576-11e9-8a76-e41e5c2ae52c.jpg)

[`DefaultImageHeaderParser#getOrientation()`](https://github.com/bumptech/glide/blob/6022f500e18fccfbe364ceabccc29d153bceb930/library/src/main/java/com/bumptech/glide/load/resource/bitmap/DefaultImageHeaderParser.java#L163) returns `UNKNOWN_ORIENTATION`.

However, Android's gallery app and my Linux image viewer apps display the orientation correctly. Here's jhead output:

```
Resolution   : 800 x 600
Orientation  : rotate 90
Jpeg process : Progressive
JPEG Quality : 93
```

So the orientation is in there, but maybe it is hard to parse?

**Comment by photospringdev** [comment]
We are having the same issue - some with Google Pixel but also with Samsung galaxy. Gallery app on our device displays correctly. 
I cannot publicly share the photos that have issues but if you email support@ photospring I can email you the photos.

**Comment by Apu-RR** [comment]
Same issue here, will this be fixed anytime soon?

**Comment by quite** [comment]
Same issue on Nokia 5.3 (2020) with Android 10. Seems consistent for all photos taken in portrait orientation.
exiv2 says: `Exif.Image.Orientation                       Short       1  right, top`
Attaching one.

![IMG_20210426_155126](https://user-images.githubusercontent.com/488679/116099762-950a5480-a6ac-11eb-9543-a3535d154ae0.jpg)

**Comment by grote** [comment]
> Glide (but in its defense also my browser) fails to determine the orientation of this image

Update: Firefox now correctly identifies the orientation. I guess it is time for glide to follow.

**Comment by quite** [comment]
> Update: Firefox now correctly identifies the orientation. I guess it is time for glide to follow.

Does the old and new Firefox behave the same with the image I posted above?

**Comment by quite** [comment]
I did manage to install Firefox 70. Both it and recent Firefox 88 displays the above photo correctly as a portrait.

**Comment by grote** [comment]
Sorry, I don't remember what Firefox version I had installed in 2019 when I reported this issue. But good to know that even Firefox 70 fixed the problem.
